src/PrestaShopBundle/Controller/Admin/Improve/International/TaxController.php

```diff
@@ -0,0 +1,279 @@
+     *     redirectRoute="admin_taxes_index",
+     *     message="You do not have permission to delete this."
+     * )
+     * @DemoRestricted(redirectRoute="admin_taxes_index")
+     *
+     * @param int $taxId
+     *
+     * @return RedirectResponse
+     */
+    public function deleteAction($taxId)
+    {
+        try {
+            $this->getCommandBus()->handle(new DeleteTaxCommand((int) $taxId));
+        } catch (TaxException $exception) {
+            $this->addFlash('error', $this->getErrorByExceptionType($exception));
+
+            return $this->redirectToRoute('admin_taxes_index');
+        }
+
+        $this->addFlash(
+            'success',
+            $this->trans('Successful deletion.', 'Admin.Notifications.Success')
+        );
+
+        return $this->redirectToRoute('admin_taxes_index');
+    }
+
+    /**
+     * Toggles status.
+     *
+     * @param int $taxId
+     *
+     * @AdminSecurity(
+     *     "is_granted('update', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_taxes_index",
+     *     message="You do not have permission to edit this."
+     * )
+     * @DemoRestricted(redirectRoute="admin_taxes_index")
+     *
+     * @return RedirectResponse
+     */
+    public function toggleStatusAction($taxId)
+    {
+        try {
+            $this->getCommandBus()->handle(new ToggleTaxStatusCommand((int) $taxId));
+        } catch (TaxException $exception) {
+            $this->addFlash('error', $this->getErrorByExceptionType($exception));
+
+            return $this->redirectToRoute('admin_taxes_index');
+        }
+
+        $this->addFlash(
+            'success',
+            $this->trans('The status has been successfully updated.', 'Admin.Notifications.Success')
+        );
+
+        return $this->redirectToRoute('admin_taxes_index');
+    }
+
+    /**
+     * Update taxes status on bulk action.
+     *
+     * @AdminSecurity(
+     *     "is_granted('update', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_taxes_index",
+     *     message="You do not have permission to edit this."
+     * )
+     *
+     * @param Request $request
+     * @param $newStatus
+     *
+     * @return RedirectResponse
+     */
+    public function bulkToggleStatusAction(Request $request, $newStatus)
+    {
+        $taxIds = $request->request->get('tax_bulk');
+        try {
+            $this->getCommandBus()->handle(
+                new BulkToggleTaxStatusCommand($taxIds, new TaxStatus($newStatus))
+            );
+        } catch (TaxException $exception) {
+            $this->addFlash('error', $this->getErrorByExceptionType($exception));
+
+            return $this->redirectToRoute('admin_taxes_index');
+        }
+        $this->addFlash(
+            'success',
+            $this->trans('The status has been successfully updated.', 'Admin.Notifications.Success')
+        );
+
+        return $this->redirectToRoute('admin_taxes_index');
```

Comment:
you can keep single redirect by moving success message inside `try` block and removing redirect from `catch`.